File: python/tracing/anthropic/02_streaming.py
"""Trace and assemble an Anthropic Messages stream."""

from respan import workflow

from _shared import (
    example_attributes,
    make_client,
    make_respan,
    model_name,
    print_result,
    workflow_name,
)

CASE_ID = "streaming"


@workflow(name=workflow_name(CASE_ID))
def run_streaming() -> str:
    with make_client().messages.stream(
        model=model_name(),
        max_tokens=64,
        messages=[
            {
                "role": "user",
                "content": "Stream one concise sentence about trace context.",
            }
        ],
    ) as stream:
        text = "".join(stream.text_stream)
        stream.get_final_message()
    return text


def main() -> None:
    respan = make_respan()
    try:
        with example_attributes(respan, CASE_ID):
            output = run_streaming()
    finally:
        respan.shutdown()
    print_result(CASE_ID, output)


if __name__ == "__main__":
    main()

File: typescript/tracing/anthropic/02_streaming.ts
import { MODEL, createRuntime, runCase } from "./_shared.js";

const caseId = "streaming";
const { client, respan } = createRuntime();

try {
  const output = await runCase(respan, caseId, async () => {
    const stream = await client.messages.create({
      model: MODEL,
      max_tokens: 80,
      stream: true,
      messages: [
        { role: "user", content: "Reply with exactly: Anthropic streaming tracing works." },
      ],
    });

    let text = "";
    for await (const event of stream) {
      if (event.type === "content_block_delta" && event.delta.type === "text_delta") {
        text += event.delta.text;
      }
    }
    return text;
  });
  console.log(JSON.stringify({ caseId, output }));
} finally {
  await respan.shutdown();
}
